test/FloatTests.res

```diff
@@ -0,0 +1,52 @@
+open RescriptCore
+
+let eq = (a, b) => a == b
+
+Test.run(__POS_OF__("clamp"), Float.clamp(4.2), eq, 4.2)
+Test.run(__POS_OF__("clamp - < min"), Float.clamp(~min=4.3, 4.1), eq, 4.3)
+Test.run(__POS_OF__("clamp - > min"), Float.clamp(~min=4.1, 4.2), eq, 4.2)
+Test.run(__POS_OF__("clamp - < max"), Float.clamp(~max=4.3, 4.2), eq, 4.2)
+Test.run(__POS_OF__("clamp - > max"), Float.clamp(~max=4.1, 4.2), eq, 4.1)
+Test.run(__POS_OF__("clamp - < min, < max"), Float.clamp(~min=4.3, ~max=4.5, 4.2), eq, 4.3)
+Test.run(__POS_OF__("clamp - < min, > max"), Float.clamp(~min=4.3, ~max=4.1, 4.2), eq, 4.3) // min wins
+Test.run(__POS_OF__("clamp - > min, < max"), Float.clamp(~min=4.1, ~max=4.5, 4.2), eq, 4.2)
+Test.run(__POS_OF__("clamp - > min, > max"), Float.clamp(~min=4.1, ~max=4.1, 4.2), eq, 4.1)
+Test.run(__POS_OF__("clamp - nan"), Float.clamp(~min=4.1, ~max=4.3, nan), eq, 4.1) // min wins ??
+Test.run(__POS_OF__("clamp - infinity"), Float.clamp(~min=4.1, ~max=4.3, infinity), eq, 4.3)
+Test.run(__POS_OF__("clamp - -infinity"), Float.clamp(~min=4.1, ~max=4.3, neg_infinity), eq, 4.1)
+Test.run(__POS_OF__("clamp - min nan"), Float.clamp(~min=nan, 4.2)->Float.isNaN, eq, true) // ??
+Test.run(__POS_OF__("clamp - max nan"), Float.clamp(~max=nan, 4.2), eq, 4.2)
+Test.run(
+  __POS_OF__("clamp - min nan, max nan"),
+  Float.clamp(~min=nan, ~max=nan, 4.2)->Float.isNaN,
+  eq,
+  true, // ??
+)
```

Comment:
I have a hard time explaining some of these `NaN` tests. Not terribly important, but I'm slightly uncomfortable with not completely understanding what's going on.

Reply:
My understanding of `NaN` is that comparing it to anything in any way should evaluate to `false`. But these results seem inconsistent with that.

Reply:
Found the problem: https://github.com/rescript-lang/rescript-compiler/issues/6040

Reply:
Fixed in https://github.com/rescript-association/rescript-core/pull/90/commits/bc4a5522b3777eb043f11df6bd00d2f30a749a87 by adding a type annotation :roll_eyes: Because the behaviour when the primitives are monomorphized and inlined is correct.